go-controller/pkg/ovn/address_set/address_set.go

```diff
@@ -271,37 +282,54 @@ func newOvnAddressSet(name string, ips []net.IP) (*ovnAddressSet, error) {
 		as.ips[ip.String()] = ip
 	}
 
-	uuid, stderr, err := util.RunOVNNbctl("--data=bare",
-		"--no-heading", "--columns=_uuid", "find", "address_set",
-		"name="+as.hashName)
+	addrSetList := &[]nbdb.AddressSet{}
+	err := as.nbClient.WhereCache(
+		func(addrSet *nbdb.AddressSet) bool {
+			return addrSet.Name == as.hashName
+		}).List(addrSetList)
 	if err != nil {
-		return nil, fmt.Errorf("find failed to get address set %q, stderr: %q (%v)",
-			as.name, stderr, err)
+		return nil, err
 	}
-	as.uuid = uuid
 
-	if uuid != "" {
+	if len(*addrSetList) > 0 {
+		// if there is already an addressSet, reuse the addressSet and set the IPs to the slice provided
+		as.uuid = (*addrSetList)[0].UUID
 		klog.V(5).Infof("New(%s) already exists; updating IPs", asDetail(as))
-		// ovnAddressSet already exists in the database; just update IPs
-		if err := as.setIPs(ips); err != nil {
+
+		var ipStrings []string
+		for _, ip := range ips {
+			ipStrings = append(ipStrings, ip.String())
+		}
+		addrset := &nbdb.AddressSet{UUID: as.uuid, Addresses: ipStrings}
```

Comment:
I guess I would take care to set the `ExternalIDs` here as well, at least to be explicit about it.

Reply:
done